lib/api_base_rest/cache/cache_manager.dart

```diff
@@ -0,0 +1,22 @@
+import "package:flutter_cache_manager/flutter_cache_manager.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+part "cache_manager.g.dart";
+
+@Riverpod(keepAlive: true)
+CacheManager cacheManagerSingleton(Ref ref) {
```

Comment:
since you're using this singleton anyway, you can inject this into `getAndCacheData` with  `ref.watch` and not as a function argument